hazelcast/src/main/java/com/hazelcast/spi/eviction/EvictionPolicyComparator.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.spi.eviction;
+
+import java.io.Serializable;
+import java.util.Comparator;
+
+/**
+ * A kind of {@link java.util.Comparator} to be
+ * used while comparing entries to be evicted.
+ *
+ * @param <K> type of the key
+ * @param <V> type of the value
+ * @param <E> type of the {@link EvictableEntryView}
+ *
+ * @see com.hazelcast.map.MapEvictionPolicyComparator
+ * @see com.hazelcast.cache.CacheEvictionPolicyComparator
+ */
+
+@SuppressWarnings("checkstyle:interfaceistype")
+public interface EvictionPolicyComparator<K, V, E extends EvictableEntryView<K, V>>
```

Comment:
Maybe also `@FunctionalInterface`? Same for `MapEvictionPolicyComparator` and `CacheEvictionPolicyComparator`.

Reply:
added annotation.